### PR TITLE
Standardize more instances of skolems

### PR DESCRIPTION
**Pull request by ajreynol**
This is work towards limiting our usage of witness terms for skolems.

It standardizes witnessing of disequalities for sets, bags, and arrays in the appropriate way that uses `mkSkolemFunction(...)` instead of `mkSkolem(...)`.

### CODE REVIEW

File: src/theory/arrays/theory_arrays.cpp [hunk 1]
```diff
@@ -1184,19 +1170,7 @@ void TheoryArrays::presolve()
 
 Node TheoryArrays::getSkolem(TNode ref)
 {
-  // the call to SkolemCache::getExtIndexSkolem should be deterministic, but use
-  // cache anyways for now
-  Node skolem;
-  std::unordered_map<Node, Node>::iterator it = d_skolemCache.find(ref);
```
Comment (mudathirmahgoub) [inline src/theory/arrays/theory_arrays.cpp:1190]: Do we still need d_skolemCache after this change?

Comment (ajreynol) [reply src/theory/arrays/theory_arrays.cpp:1190]: No, it is deleted with this PR.

Comment (mudathirmahgoub) [reply src/theory/arrays/theory_arrays.cpp:1190]: Yes, I missed that.

**Review by mudathirmahgoub — APPROVED** [review]
LGTM. The code is more clean now.

**Pull request merged**